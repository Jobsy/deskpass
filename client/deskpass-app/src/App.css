
.app {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-family: sans-serif;
  height: 100vh;
}
