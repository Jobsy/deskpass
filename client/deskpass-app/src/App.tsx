
import styled from "styled-components";

import StripQuote from "./components/StripQuote";
import ToolTip from "./components/ToolTip";
import "./App.css";


const TooltipTarget = styled.span``;

const App = (props: any) => {
  return (
    <div data-testid="app" className="app">
      <div>
        <ToolTip tooltipText="I am a left tooltip" position="left" background="00ADB5">
          <TooltipTarget>Left</TooltipTarget>
        </ToolTip>
        <ToolTip tooltipText="I am a top tooltip" position="top" background="FF8C00">
          <TooltipTarget>Top</TooltipTarget>
        </ToolTip>
        <ToolTip tooltipText="I am a bottom tooltip" position="bottom" background="03AC13">
          <TooltipTarget>Bottom</TooltipTarget>
        </ToolTip>
        <ToolTip tooltipText="I am a right tooltip" position="right" background="9897A9">
          <TooltipTarget>Right</TooltipTarget>
        </ToolTip>
      </div>

      <StripQuote />
    </div>
  );
}

export default App;