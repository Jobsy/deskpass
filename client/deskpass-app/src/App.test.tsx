
import { cleanup, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer"
import App from "./App";


afterEach(() => {
    cleanup();
})

test('should render StripQuote component', () => {
    render(<App />);
    const appComponent = screen.getByTestId("app")
    
    expect(appComponent).toBeInTheDocument();
});

test('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON()

    expect(tree).toMatchSnapshot()
});

export {}