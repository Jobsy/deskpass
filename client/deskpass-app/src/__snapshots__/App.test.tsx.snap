// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="app"
  data-testid="app"
>
  <div>
    <div
      className="sc-hKMtZM IMoHK"
      data-testid="tool-tip"
    >
      <button
        className="sc-eCYdqJ byKboX"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="sc-iBkjds"
        >
          Left
        </span>
      </button>
    </div>
    <div
      className="sc-hKMtZM IMoHK"
      data-testid="tool-tip"
    >
      <button
        className="sc-eCYdqJ byKboX"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="sc-iBkjds"
        >
          Top
        </span>
      </button>
    </div>
    <div
      className="sc-hKMtZM IMoHK"
      data-testid="tool-tip"
    >
      <button
        className="sc-eCYdqJ byKboX"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="sc-iBkjds"
        >
          Bottom
        </span>
      </button>
    </div>
    <div
      className="sc-hKMtZM IMoHK"
      data-testid="tool-tip"
    >
      <button
        className="sc-eCYdqJ byKboX"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="sc-iBkjds"
        >
          Right
        </span>
      </button>
    </div>
  </div>
  <div
    className="sc-bczRLJ eJKZJo"
    data-testid="stripe-quote"
  >
    <div
      className="sc-hKMtZM IMoHK"
      data-testid="tool-tip"
    >
      <button
        className="sc-eCYdqJ jWFya"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          className="sc-eCYdqJ jWFya"
        >
          <h2>
            Word-Quote Striper
          </h2>
        </button>
      </button>
    </div>
    <textarea
      autoComplete="on"
      className="sc-gsnTZi gKEenm"
      onChange={[Function]}
      placeholder="Enter quote"
      value=""
    />
    <p
      className="sc-dkzDqf ejhFWH"
    >
      Invalid!
       
    </p>
  </div>
</div>
`;
